infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/logreader/RoutingQueuesLogEntryReader.java

```diff
@@ -24,9 +37,33 @@ public RoutingQueuesLogEntryReader(CorfuRuntime runtime, LogReplicationSession s
     }
 
     @Override
-    public LogReplicationEntryMsg read(UUID logEntryRequestId) throws TrimmedException {
-        // Reads from the queue corresponding to this destination.  The queue contains addresses in the main queue
-        // from where the actual data(payload) is to be read.
-        return null;
+    OpaqueEntry filterTransactionEntry(OpaqueEntry opaqueEntry) {
+        List<SMREntry> routingTableEntryMsgs = opaqueEntry.getEntries().get(LogReplicationUtils.lrLogEntrySendQId);
+
+        List<SMREntry> filteredMsgs = new ArrayList<>();
+
+        for (SMREntry entry : routingTableEntryMsgs) {
+            Object[] objs = entry.getSMRArguments();
+            ByteBuf valueBuf = Unpooled.wrappedBuffer((byte[])objs[1]);
+            RoutingTableEntryMsg msg =
+                    (RoutingTableEntryMsg)replicationContext.getProtobufSerializer().deserialize(valueBuf, null);
+
+            if (msg.getDestinationsList().contains(session.getSinkClusterId())) {
+                filteredMsgs.add(entry);
+            }
+        }
+        HashMap<UUID, List<SMREntry>> opaqueEntryMap = new HashMap<>();
+        opaqueEntryMap.put(CorfuRuntime.getStreamID(REPLICATED_QUEUE_NAME_PREFIX + session.getSourceClusterId()),
+                filteredMsgs);
+        return new OpaqueEntry(opaqueEntry.getVersion(), opaqueEntryMap);
+    }
+
+    @Override
+    boolean isValidTransactionEntry(@NonNull OpaqueEntry entry) {
+        Set<UUID> txEntryStreamIds = new HashSet<>(entry.getEntries().keySet());
+        Preconditions.checkState(txEntryStreamIds.size() == 1, "Routing queue session's log" +
+                " entries should only come from the shared data queue for log entry sync");
+
+        return txEntryStreamIds.contains(LogReplicationUtils.lrLogEntrySendQId);
```

Comment:
Precondition check throws an exception if it is not met, which will result in log entry sync being cancelled.  You can replace this with a simple check and return true or false accordingly.